engine/src/main/java/com/google/android/fhir/sync/Sync.kt

```diff
@@ -108,10 +126,89 @@ object Sync {
           }
       }
 
+  /**
+   * Combines the sync state for a periodic sync operation, including work state, progress, and
+   * terminal states.
+   *
+   * @param context The Android application context.
+   * @param workName The name of the periodic sync work.
+   * @param syncJobProgressStateFlow A flow representing the progress of the sync job.
+   * @return A flow of [PeriodicSyncState] combining the sync job states.
+   */
+  @PublishedApi
+  internal fun combineSyncStateForPeriodicSync(
+    context: Context,
+    workName: String,
+    syncJobProgressStateFlow: Flow<SyncJobStatus>,
+    id: UUID,
+  ): Flow<PeriodicSyncState> {
+    val workStateFlow: Flow<WorkInfo.State> = observeWorkState(context, id)
+    val syncJobTerminalStateFlow: Flow<SyncJobStatus> =
+      FhirEngineProvider.getFhirDataStore(context).observeSyncJobTerminalState(workName)
+    val syncJobFlow =
+      combine(syncJobProgressStateFlow, syncJobTerminalStateFlow) {
+        inProgressStatus,
+        terminalStatus,
+        ->
+        if (terminalStatus == SyncJobStatus.Unknown) {
+          inProgressStatus
+        } else {
+          terminalStatus
+        }
+      }
+
+    return combine(workStateFlow, syncJobFlow) { workInfoState, syncJobStatus ->
+      val lastSyncJobStatus =
+        FhirEngineProvider.getFhirDataStore(context).getLastSyncJobStatus(workName)
+      val currentJobState = createCurrentJobState(syncJobStatus, workInfoState)
+      PeriodicSyncState(
+        lastJobState = mapSyncJobStatusToResult(lastSyncJobStatus),
+        currentJobState = currentJobState,
+      )
+    }
+  }
```

Comment:
you can simplify this function now:

```suggestion
  internal fun combineSyncStateForPeriodicSync(
    context: Context,
    workName: String,
    workerInfoSyncJobStatusPairFromWorkManagerFlow: Flow<Pair<WorkInfo.State, SyncJobStatus?>>,
  ): Flow<PeriodicSyncState> {
    val syncJobStatusInDataStoreFlow: Flow<SyncJobStatus?> =
      FhirEngineProvider.getFhirDataStore(context).observeTerminalSyncJobStatus(workName)

    return combine(workerInfoSyncJobStatusPairFromWorkManagerFlow, syncJobStatusInDataStoreFlow) {
      workerInfoSyncJobStatusPairFromWorkManager,
      syncJobStatusFromDataStore,
      ->
      PeriodicSyncState(
        lastJobState = mapSyncJobStatusToResult(syncJobStatusFromDataStore),
        currentJobState =
          createSyncState(
            workerInfoSyncJobStatusPairFromWorkManager.first,
            workerInfoSyncJobStatusPairFromWorkManager.second,
            syncJobStatusFromDataStore,
          ),
      )
    }
  }
```

Reply:
done.